Mathlib/RingTheory/PowerSeries/Order.lean

```diff
@@ -226,18 +229,50 @@ theorem coeff_mul_prod_one_sub_of_lt_order {R ι : Type*} [CommRing R] (k : ℕ)
     rw [Finset.prod_insert ha, ← mul_assoc, mul_right_comm, coeff_mul_one_sub_of_lt_order _ t.1]
     exact ih t.2
 
--- TODO: link with `X_pow_dvd_iff`
-theorem X_pow_order_dvd (h : order φ < ⊤) : X ^ (order φ).lift h ∣ φ := by
-  refine ⟨PowerSeries.mk fun n => coeff R (n + (order φ).lift h) φ, ?_⟩
+/-- Given a non-zero power series `f`, `divXPowOrder f` is the power series obtained by
+  dividing out the largest power of X that divides `f`, that is its order -/
+def divXPowOrder (f : R⟦X⟧) : R⟦X⟧ :=
+  .mk fun n ↦ coeff R (n + f.order.toNat) f
+
+@[deprecated (since := "2025-04-15")]
+noncomputable alias divided_by_X_pow_order := divXPowOrder
+
+@[simp]
+lemma coeff_divXPowOrder {f : R⟦X⟧} {n : ℕ} :
+    coeff R n (divXPowOrder f) = coeff R (n + f.order.toNat) f :=
+  coeff_mk _ _
+
+@[simp]
+lemma divXPowOrder_zero :
+    divXPowOrder (0 : R⟦X⟧) = 0 := by
+  ext
+  simp
+
+lemma constantCoeff_divXPowOrder {f : R⟦X⟧} :
+    constantCoeff R (divXPowOrder f) = coeff R f.order.toNat f := by
+  simp [← coeff_zero_eq_constantCoeff]
+
+lemma constantCoeff_divXPowOrder_eq_zero_iff {f : R⟦X⟧} :
+    constantCoeff R (divXPowOrder f) = 0 ↔ f = 0 := by
+  by_cases h : f = 0
+  · simp [h]
+  · simp [constantCoeff_divXPowOrder, coeff_order h, h]
+
+theorem self_eq_X_pow_order_mul_divXPowOrder {f : R⟦X⟧} :
+    X ^ f.order.toNat * divXPowOrder f = f := by
```

Comment:
I'd say this name doesn't quite sense any more, since it's not of the form `self_eq_`! Perhaps `X_pow_order_mul_divXPowOrder`?

Reply:
Actually it never was the right way around, but I agree, let's change it (plus it makes the name shorter).